packages/shared/src/api/lanyardApi.ts

```diff
@@ -60,15 +60,25 @@ export async function checkAttestedSignature(signData: string) {
   return false;
 }
 
-export async function discoverContacts(phoneNums: string[]) {
+export async function discoverContacts(
+  phoneNums: string[]
+): Promise<[string, string][]> {
   try {
-    const nums = diffContactBook(phoneNums);
-    // const lastSalt = formatUw('0');
-    const payload = ['whose-bulk', 0, nums.last, nums.add, nums.del];
-    const noun = dwim(payload);
-
+    const nums = await diffContactBook(phoneNums);
```

Comment:
should we short circuit if there's no difference since the last request?

Reply:
Except new numbers might have gotten verified, so you'll want to just send the query anyway, no?

Reply:
Missed this earlier, will update.